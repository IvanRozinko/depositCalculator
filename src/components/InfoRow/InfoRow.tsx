import React from 'react'
import { Box, Typography, makeStyles } from '@material-ui/core';

interface Props {
  label: string,
  value: string,
  valueBold?: boolean,
  children?: React.ReactNode,
  centered?: boolean
}

function InfoRow({ label, value, valueBold, children, centered }: Props) {
  const useStyle = makeStyles({
    label: {
      display: 'flex',
      flex: '1',
      alignItems: 'center',
      fontWeight: valueBold ? 600 : 400,
      justifyContent: centered ? 'center' : 'start'
    }
   })
  const styles = useStyle(); 
  return (
    <Box
      display='flex'
      flexDirection='row'
    >
      <Typography
        className={styles.label}
      >
        {label}
        {children}
      </Typography>
      <Typography
        className={styles.label}
      >
        {value}
      </Typography>
    </Box>
  )
}

export default InfoRow
