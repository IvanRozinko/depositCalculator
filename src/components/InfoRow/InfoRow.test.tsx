import InfoRow from "./InfoRow";
import { shallow } from 'enzyme';

describe('<InfoRow />:', () => {
  const wrapper = shallow(
    <InfoRow 
      label='label'
      value='value'
    />
  )

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should has label and value', () => {
    expect(wrapper.text().includes('label')).toBe(true);
    expect(wrapper.text().includes('value')).toBe(true);
  })
})