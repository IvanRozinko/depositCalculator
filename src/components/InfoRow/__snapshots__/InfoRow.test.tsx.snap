// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoRow />: should render correctly 1`] = `
<Styled(MuiBox)
  display="flex"
  flexDirection="row"
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-label-1"
  >
    label
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-label-1"
  >
    value
  </WithStyles(ForwardRef(Typography))>
</Styled(MuiBox)>
`;
